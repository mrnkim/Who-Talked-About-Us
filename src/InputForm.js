import React from "react";
import { Container, Row, Col, Button, Form } from "react-bootstrap";

function InputForm({ handleSubmit, handleChange, input, type, buttonText }) {
  return (
    <Container className="mt-5">
      <Row style={{ display: "flex", justifyContent: "center" }}>
        <Col sm={12} md={8} lg={8}>
          <Form className="d-flex" onSubmit={handleSubmit}>
            <Form.Control
              type={type}
              id={type}
              name={type}
              placeholder={type}
              onChange={handleChange}
              value={input}
              aria-label={type}
              style={{ width: "60%", fontSize: "1em" }}
            />
            <Button
              style={{ marginLeft: "3px" }}
              className="rounded"
              variant="primary"
              type="submit"
            >
              {buttonText}
            </Button>
          </Form>
        </Col>
      </Row>
    </Container>
  );
}

export default InputForm;
