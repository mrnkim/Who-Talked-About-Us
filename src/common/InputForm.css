.inputContainer {
    display: flex;
    align-items: center;
    width: 100%;
  }

  .primaryButton {
    display: flex;
    min-height: 4rem;
    padding-top: 12px;
    background: #FFF;
    border: none;
    border-bottom: 2px solid #D4D5D2;
    justify-content: center;
    align-items: center;
    transition: border-color 0.3s;
    margin-top: 0.75rem;
    margin-bottom: 0.625rem;
    height: 3.5rem;
  }

  .textField {
    display: flex;
    min-width: 27rem;
    min-height: 4rem;
    padding-top: 12px;
    flex-direction: column;
    text-wrap:wrap;
    gap: 0.625rem;
    background: #FFF;
    border: none;
    border-bottom: 2px solid #D4D5D2;
    transition: border-color 0.3s;
    margin-top: 0.75rem;
    margin-bottom: 0.625rem;
    font-size: 1.2rem;
    height: 3.5rem;
  }

  .searchTextField {
    display: flex;
    width: 100%;
    max-width: 75rem;
    min-height: 4rem;
    padding-top: 12px;
    flex-direction: column;
    text-wrap:wrap;
    gap: 0.625rem;
    background: #FFF;
    border: none;
    border-bottom: 2px solid #D4D5D2;
    transition: border-color 0.3s;
    margin-top: 0.75rem;
    margin-bottom: 0.625rem;
    font-size: 1.2rem;
    height: 3.5rem;
  }

  .inputForm {
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .inputForm:hover {
    flex-shrink: 0;
    background: var(--TL-Grey--100, #F4F4F3);
  }

  .inputFocused .primaryButton,
  .inputFocused .textField {
    border-bottom: 2px solid #9AED59;
  }

  .textField:focus {
    outline: none;
  }

  .icon {
    margin-right: 0.3rem;
  }
