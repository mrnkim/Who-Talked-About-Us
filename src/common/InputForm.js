import React from "react";
import { useState } from "react";
import "./InputForm.css";

/** User input form
 *
 * { IndexForm, ExistingIndexForm, SearchForm } -> InputForm
 *
 */

function InputForm({
  handleSubmit,
  handleChange,
  value,
  type,
  icon,
  className,
}) {
  const [isInputFocused, setInputFocused] = useState(false);

  const handleInputFocus = () => {
    setInputFocused(true);
  };

  const handleInputBlur = () => {
    setInputFocused(false);
  };

  return (
    <form className="inputForm" onSubmit={handleSubmit}>
      <div className={`inputContainer ${isInputFocused ? "inputFocused" : ""}`}>
        <button className="primaryButton" type="submit">
          {icon && <img src={icon} alt="Icon" className="icon" />}
        </button>
        <input
          type={type}
          id={type}
          name={type}
          placeholder={type}
          onChange={handleChange}
          value={value}
          aria-label={type}
          data-testid="search-input"
          className={className}
          onFocus={handleInputFocus}
          onBlur={handleInputBlur}
        />
      </div>
    </form>
  );
}

export default InputForm;
