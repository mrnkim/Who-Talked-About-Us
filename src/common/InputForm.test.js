import React from "react";
import { render } from "@testing-library/react";
import InputForm from "./InputForm.js";

it("renders without crashing", function () {
  render(<InputForm />);
});

it("matches snapshot", function () {
  const { container } = render(<InputForm />);
  expect(container).toMatchSnapshot();
});

it("has the correct element and text", function () {
  const { container } = render(<InputForm />);

  const button = container.querySelector("Button");
  expect(button.getAttribute("type")).toEqual("submit");
});
