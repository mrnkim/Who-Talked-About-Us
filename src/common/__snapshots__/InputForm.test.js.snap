// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <form
    class="d-flex justify-content-center align-items-center inputForm"
  >
    <div
      class="inputContainer "
    >
      <button
        class="primaryButton"
        type="submit"
      />
      <input
        class="textField"
        data-testid="search-input"
        value=""
      />
    </div>
  </form>
</div>
`;
